### PR TITLE
upgrades to run ES 2

### PR DESCRIPTION
**Pull request by dermidgen**
# 2.0.0
  * Upgrades to Elasticsearch 2.0.0
  * Upgrades to Kubernetes Plugin 2.0.2
  * Runs under elasticsearch user
  * Set ulimit to support memlock
  * Disables Security Manager to enable Kubernetes plugin
    - access to networking
    - access to filesystem for secrets and configuration paths

### CODE REVIEW

**Comment by dermidgen** [comment]
Added a binding to all network interfaces.

**Comment by pires** [comment]
hi @dermidgen I don't know why but Github seems to be borked regarding PRs. I never got a notification regarding your PR - and another one in another project. I went ahead and fixed it. I'm so, so sorry :(

**Comment by pires** [comment]
@dermidgen would you do this work and update the official Kubernetes example for Elasticsearch? :)

**Comment by dermidgen** [comment]
@pires - No worries at all - we're all busy people and systems ain't perfect :)
I would be happy to I would be happy to help. What "work" are you referring to specifically - updating the readme and examples? If so I definitely plan to. My next step was to work this in with https://github.com/pires/kubernetes-elasticsearch-cluster for 2.0.

Just point me at whatever work you would like me to help with; hopefully between tonight and tomorrow I should have things mostly buttoned up and ready to fully update the official Kubernetes examples with some 2.0 notes.

BTW - there's still an outstanding PR on https://github.com/pires/docker-elasticsearch for binding on all interfaces - as was pointed out in the thread on the kubernetes plugin. I also applied that in this PR since we overwrite `run.sh` from `docker-elasticsearch:2.0.0`

**Comment by dermidgen** [comment]
So I merged from your master - I guess we were doing some things in parallel.  I merged in favor of your `network.host` settings since I'm guessing you actually looked things up.  I was setting to `0.0.0.0`;

File: elasticsearch.yml [hunk 1]
```diff
@@ -23,5 +23,5 @@ cloud:
     service: ${DISCOVERY_SERVICE}
     namespace: ${NAMESPACE}
 discovery:
-    type: kubernetes
-    zen.ping.multicast.enabled: false
+  type: io.fabric8.elasticsearch.discovery.k8s.K8sDiscoveryModule
+  zen.ping.multicast.enabled: false
```
Comment (pires) [inline elasticsearch.yml]: `type: kubernetes` is how it's used, right?

File: README.md [hunk 2]
```diff
@@ -10,6 +10,23 @@ Ready to use lean (207MB) Elasticsearch Docker image ready for using within a Ku
 * Elasticsearch 2.0.0
 * Kubernetes plug-in 2.0.2
 
+## Pre-requisites
+
+* Docker 1.7.0+
```
Comment (pires) [inline README.md]: Can you change this to:
```
## Tested with
```
and remove `Docker 1.7.0+` as it's always part of Kubernetes?

File: README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # docker-elasticsearch-kubernetes
 
-Ready to use lean (207MB) Elasticsearch Docker image ready for using within a Kubernetes cluster.
+Ready to use lean (210MB) Elasticsearch Docker image ready for using within a Kubernetes cluster.
```
Comment (pires) [inline README.md]: The size is actually 207.8MB but I always miss the fraction part.

**Comment by pires** [comment]
@dermidgen made some comments. After that is taken care of, I can merge.

Regarding `kubernetes-elasticsearch-cluster`, I already provided a PR last night. Going to merge as soon as someone else besides me tests it. Now, I was wondering if you could update kubernetes/examples/elasticsearch` documentation to match the aforementioned PR?

File: elasticsearch.yml (hunk 1 above)
Comment (dermidgen) [inline elasticsearch.yml]: Oooooh bad merge sorry!

Comment (dermidgen) [inline elasticsearch.yml]: Not sure what you're asking about zen - should that be removed?

File: README.md (hunk 3 above)
Comment (dermidgen) [inline README.md]: How bout I round up?

**Comment by dermidgen** [comment]
Okay made some updates there - just wondering on the `zen` discovery comment and I think we're good.  Will start working on the docs for `kubernetes-elasticsearch-cluster`

**Comment by dermidgen** [comment]
Also - yes, I'm working on `kubernetes/examples/elasticsearch` docs as well.  I will bring over the `.yaml` config files and examples from `kubernetes-elasticsearch-cluster`.

I think this is where the current examples are confusing, because the current example provides a simple Kubernetes config that has our instance running mixed services - which barfs when you run the `kubectl scale`. I'll get it all sorted out today :)

**Comment by pires** [comment]
@dermidgen the official examples has both a simple version and a `production` version. The documentation explicitly states that.

**Comment by dermidgen** [comment]
@pires So - I'm currently working on updates to the simple version. Everything goes fine until the instructions say - great now you've got 1 node up; let's scale. Then the instructions call to run `kubectl scale --replicas=3 rc es`. This is where we break down, because the simple example uses a mixed service node.

The simple example shows that you should see something like
```
[2015-08-30 10:08:02,517][INFO ][cluster.service          ] [Hammerhead] added {[Tenpin][2gv5MiwhRiOSsrTOF3DhuA][es-78e0s][inet[/10.244.54.4:9300]]{master=true},}, reason: zen-disco-receive(join from node[[Tenpin][2gv5MiwhRiOSsrTOF3DhuA][es-78e0s][inet[/10.244.54.4:9300]]{master=true}])
```

But instead we get the unicast errors. I understand this is because our example svc and rc run mixed services.

Should we remove the scaling part of the simple example and refer folks to the production example - or is there a way to fix the simple example svc & rc so that scaling our single node works properly as a mixed service node?

**Comment by dermidgen** [comment]
@pires I think it might make better sense for me to start a PR on Kubernetes - and then continue this discussion on there where you can give line-by-line feedback?  Agree?

**Comment by dermidgen** [comment]
I went ahead a did one: https://github.com/kubernetes/kubernetes/pull/17570

**Comment by dermidgen** [comment]
@pires - I've made the required updates in the official examples.  The production example was pretty smooth. Nothing really wrong there. I was able to make the updates and follow the guide and everything went swimmingly.

A couple things:
 * The _simple_ example instructions have stuff about scaling - I think we should remove that
 * The warnings about `emptyDir` - I'm not sure what to do with that. We have to remove that for Kubernetes 1.1+ but does that change the spirit of the warning?

**Comment by dermidgen** [comment]
Oh N00z!

File: elasticsearch.yml [hunk 4]
```diff
@@ -16,12 +16,16 @@ path:
 
 bootstrap.mlockall: true
 
-http.enabled: ${HTTP_ENABLE}
+http:
+  enabled: ${HTTP_ENABLE}
+  compression: true
+  cors:
+    enabled: true
```
Comment (dermidgen) [inline elasticsearch.yml]: @pires - super stoked on this by the way

Comment (pires) [inline elasticsearch.yml]: Awesome!

**Comment by pires** [comment]
Most of this PR was merged with cherry-pick. Thank you @dermidgen

**Pull request closed without merging**